run_blockchain_terraformer.py

```diff
@@ -466,6 +468,10 @@ def organize_pool_details_uni_v3(
         return None
     pair = pair[:-1]
     fee = pool_data["args"]["fee"]
+    fee_float = float(fee) / 1000000
+    if exchange == NILE_V3_NAME:
+        pool = web3.eth.contract(address=web3.to_checksum_address(pool_address), abi=NILE_V3_POOL_ABI)
+        fee_float = pool.caller.currentFee() / 1000000
```

Comment:
Why function `currentFee` and not function `fee`?
For UniV3 pools, we are using function `fee`.
Might be worth verifying exactly which function is required here.
You can do that by looking into the source code of this pool contract, checking which fee is used in the swap function, and then following it to see which of these two functions returns that specific fee.

Reply:
And why you using `1000000` in some cases and `1e6` in other cases?

Reply:
Also, creating a `web3.eth.contract` instance clearly doesn't belong in here (as evident by how it is done elsewhere in the code. You can't just "throw stuff in" without taking into consideration the broader SW design and structure.

Reply:
Also, it doesn't make sense that we need to make an RPC at this point, as the fee has already been obtained previously.
So during that previous point - either it was already obtained correctly using this contract function, or it there was an attempt to use the other function, which would have reverted for this contract.

Reply:
1. The 1000000 vs 1e6 is just poor consistency - I will update this to use 1e6.
2. Nile V3 uses a fee override function - currentFee. This results in an awkward situation in the Terraformer, as in all other cases the bot obtains the fee from the pool creation event. Due to the override, the best way to obtain the pool's current fee is by directly calling the contract. 
3. Regarding the overall SW design, do you have a suggestion for a more appropriate place for this, given that it's only relevant for Nile V3? I'm happy to move it if it makes more sense elsewhere.

Reply:
Please see the PR spawned off of this one.
In order to handle this in the uniswap_v3 module, for example, I used a paradigm similar to the one implemented in the solidly_v2 module.

The part:
```
        pool = web3.eth.contract(address=web3.to_checksum_address(pool_address), abi=NILE_V3_POOL_ABI)
        fee_float = pool.caller.currentFee() / 1000000
```
Doesn't look like it belongs here.
Even without being acquainted with this module, a simple observation reveals that no other DEX requires an RPC for fetching the fee at this point, so why should the Nile V3 DEX require one?
Subsequently, placing it here breaks the structure of the code and the over design of this module.
Wherever the fee for all the other DEXs is fetched, that's where the fee for Nile V3 should be fetched.